### PR TITLE
Add One Bonk KO

### PR DESCRIPTION
**Pull request by mracsys**
When enabled, if Link collides and recoils after rolling into a wall or object, or "bonking", he keels over dead.

Repurposes Link's state3 flag 4 to handle bonk animation cancellation. Per engineer124, It does not appear to be used anywhere beyond getting set and unset.

<s>Marking as draft until animation cancellation is handled. Current implementation sets Link's health to 0 on the last frame of the bonk animation. Gaps include:

- using ocarina at any point during the bonk will interrupt it and not ko you
- bonking while hovering/in air will not ko you (falling cancels it out)
- removing irons mid-bonk while under water will not ko you (swimming cancels it out)
- bonking into chests while opening them will not ko you
- receiving damage during bonk will either interrupt or cancel it out and not ko you</s>

### CODE REVIEW

File: data/World/Water Temple MQ.json [hunk 1]
```diff
@@ -65,7 +65,7 @@
         "dungeon": "Water Temple",
         "locations": {
             "Water Temple MQ Freestanding Key": "
-                Hover_Boots or can_use(Scarecrow) or logic_water_north_basement_ledge_jump",
+                Hover_Boots or can_use(Scarecrow) or logic_water_north_basement_ledge_jump or can_bonk_crate",
```
Comment (dewiniaid) [inline data/World/Water Temple MQ.json:68]: I'm by no means an expert on logic (and especially not MQ logic), but this change looks suspicious.  Was this `or` meant to be an `and`, possibly grouped with one or more of the other options?

Comment (mracsys) [reply data/World/Water Temple MQ.json:68]: yes, should be an and

File: data/settings_mapping.json [hunk 2]
```diff
@@ -340,7 +340,8 @@
             "misc_hints",
             "text_shuffle",
             "damage_multiplier",
-			"no_collectible_hearts",
+            "deadly_bonks",
+      			"no_collectible_hearts",
```
Comment (fenhl) [inline data/settings_mapping.json:344]: might as well completely fix the indentation here (replace tabs with spaces)

File: data/World/Water Temple MQ.json [hunk 3]
```diff
@@ -65,7 +65,7 @@
         "dungeon": "Water Temple",
         "locations": {
             "Water Temple MQ Freestanding Key": "
-                Hover_Boots or can_use(Scarecrow) or logic_water_north_basement_ledge_jump or can_bonk_crate",
+                ((Hover_Boots or can_use(Scarecrow) or logic_water_north_basement_ledge_jump) and can_bonk_crate)",
```
Comment (dewiniaid) [inline data/World/Water Temple MQ.json:68]: Being overly pedantic but: outer set of parenthesis are unnecessary here.

**Comment by fenhl** [comment]
Since all the reasons for marking as a draft seem to be handled, is this ready to review now?

**Comment by fenhl** [comment]
This setting should probably be on in the Hell Mode preset.

**Comment by mracsys** [comment]
It's ready for review. Only thing keeping it in draft is whether or not this should be part of a new Memes/Gimmicks/Use-At-Your-Own-Risk tab.

**Comment by cjohnson57** [comment]
I just commented this in another PR, but I'm not a fan of the idea of a memes tab

**Comment by r0bd0g** [comment]
IMO we either need a memes tab or to start being more choosy about what weird features we allow.

File: data/World/Overworld.json [hunk 4]
```diff
@@ -1016,11 +1016,11 @@
             "Sheik in Kakariko": "
                 is_adult and Forest_Medallion and Fire_Medallion and Water_Medallion",
             "Kak Anju as Adult": "is_adult and at_day",
-            "Kak Anju as Child": "is_child and at_day",
+            "Kak Anju as Child": "is_child and at_day and can_bonk_crate",
```
Comment (fenhl) [inline data/World/Overworld.json:1019]: This should be in logic without the crate if 6 or fewer cuccos are required. I'm not sure if `chicken_count < 7` will work as an expression though.

**Comment by r0bd0g** [comment]
I think crates that have skulltulas inside of them (like in mq deku or mq water, not like the crates where they 'pop' out), it needs to be a trick to kill them and collect the tokens that pop out the top. Precedent for this is the skulltula in the rubble in mq shadow needs explosives.

**Comment by r0bd0g** [comment]
There are some silver Rupees inside crates in MQ DC.

File: data/World/Overworld.json (hunk 4 above)
Comment (fenhl) [reply data/World/Overworld.json:1019]: Just checked and can confirm that it works as expected.

Comment (mracsys) [reply data/World/Overworld.json]: I added this in, thanks!

File: data/World/Water Temple MQ.json (hunk 3 above)
Comment (mracsys) [reply data/World/Water Temple MQ.json]: Updated

**Comment by mracsys** [comment]
VC crashing in fixed camera scenes is fixed, and logic is updated. I did not add tricks for the skulltulas. I changed my mind on the shadow skull change for shortcuts, that should probably be default.

The player should not be able to dodge death with health dropping on them from a crate, but I have not been able to confirm it. This will not be relevant until/if pot and crate shuffle is merged.

Since the meme tab has been shot down and this feature has full logic developed, it should be ready to review now with the setting living on the Other tab.

**Review by cjohnson57 — APPROVED** [review]
Looks good to me.

FYI, I'm going to merge r0b's logic updates first, so there may be merge conflicts to deal with before I merge this.

**Pull request merged**